docs/custom_build.md

```diff
@@ -102,23 +102,44 @@ There are a couple of caveats you should be aware of with `buildah` (and similar
 
 All the commands above contain `$EXPECTED_REF`. What is that?
 
-### Tags
+### Build Tags
```

Comment:
Imo you can hammer home the difference btwn build and deploy tags a bit, since i suspect ppl reading this won't initially realize they're different. consider either a section called "image tags" / "build and deploy tags" or two separate sections, one for "Build tags" and one for "deploy tags".

I also might explain "because different built tools..." first and then later differentiate: "regardless of the tag it BUILT with, Tilt always DEPLOYS with..."

Reply:
I ended up rewriting this whole section. I was trying to preserve as much as I could, but the more I read over it, the more I saw different things that were subtly wrong or confusing. Even the way this section calls them different types of refs is weird and inaccurate.